### PR TITLE
control/controlhttp: use secure WebSockets protocol by default

### PR DESCRIPTION
**Pull request by mihaip**
Forcing the insecure protocol (and perserving the port number) is only
desired for localhost testing, in prod we need to use wss:// to avoid
mixed-content errors.

Signed-off-by: Mihai Parparita <mihai@tailscale.com>

### CODE REVIEW

File: control/controlhttp/client_js.go [hunk 1]
```diff
@@ -28,9 +28,15 @@ func Dial(ctx context.Context, addr string, machineKey key.MachinePrivate, contr
 	if err != nil {
 		return nil, err
 	}
+	wsScheme := "wss"
+	wsHost := host
+	if host == "localhost" {
+		wsScheme = "ws"
+		wsHost = net.JoinHostPort(host, addr)
```
Comment (bradfitz) [inline control/controlhttp/client_js.go:35]: Line 27 makes this super confusing... addr changes from a host:port to just a port. Rename the variable on line 27?

Comment (mihaip) [reply control/controlhttp/client_js.go]: Yeah, not sure what I was thinking there. Renamed that to `port`

File: control/controlhttp/client_js.go [hunk 2]
```diff
@@ -16,21 +16,27 @@ import (
...
-	host, addr, err := net.SplitHostPort(addr)
+	host, port, err := net.SplitHostPort(addr)
 	if err != nil {
 		return nil, err
 	}
+	wsScheme := "wss"
+	wsHost := host
+	if host == "localhost" {
+		wsScheme = "ws"
+		wsHost = net.JoinHostPort(host, port)
```
Comment (bradfitz) [inline control/controlhttp/client_js.go:35]: so now this is just `wsHost = addr` ?

File: control/controlhttp/client_js.go [hunk 3]
```diff
@@ -16,21 +16,27 @@ import (
...
 		return nil, err
 	}
 
-	host, addr, err := net.SplitHostPort(addr)
+	host, port, err := net.SplitHostPort(addr)
 	if err != nil {
 		return nil, err
 	}
+	wsScheme := "wss"
+	wsHost := host
```
Comment (bradfitz) [inline control/controlhttp/client_js.go:32]: and this is just to make 80 into 443, right?

related: https://github.com/tailscale/tailscale/pull/4323

cc @maisem

Comment (bradfitz) [reply control/controlhttp/client_js.go:32]: nevermind, this is client_js.go

**Review by bradfitz — APPROVED** [review]

File: control/controlhttp/client_js.go (hunk 2 above)
Comment (mihaip) [reply control/controlhttp/client_js.go]: True, we don't actually need to parse out `port`. Simplified further.

File: control/controlhttp/client_js.go (hunk 3 above)
Comment (mihaip) [reply control/controlhttp/client_js.go:32]: I think client_js.go could be folded into client.go (as a third fallback to HTTP/HTTPS) if we wanted to do that bug and/or tailscale/corp#5542.

**Pull request merged**